Functions.py

```diff
@@ -79,8 +79,11 @@ def calc_regional_sources(
         sources_per_region = sources.weighted(weights).sum(dim=(lat_name, lon_name))
     else:
         mask_3D = regions.mask_3D(sources, lon_name=lon_name, lat_name=lat_name)
-        weights = np.cos(np.deg2rad(sources[lat_name]))
-        weights = (mask_3D * weights).fillna(0)
+        # Next lines is not needed if we use srcs_frac, as they are already area-weighted results
+        # The weights (cosine factors) can be substituted by 1.
+        #weights = np.cos(np.deg2rad(sources[lat_name]))
+        #weights = (mask_3D * weights).fillna(0)
+        weights = (mask_3D*1.).fillna(0)
```

Comment:
For someone looking at this code later on (ourselves in a few months, or someone else after we publish the code together with the paper), I think it makes more sense to explain the final code, rather than the changes in this PR. If someone wants to understand how it came to this, they can look back at this PR to see the history and the discussion around the changes. 

Also, couldn't we simply skip the multiplication by 1 altogether?

```suggestion
        # Use equal weights for each region as we are already using area-weighted fractional sources.
        # Effectively, the weighting is only used to group grid points of the same region together.
        weights = mask_3D.fillna(0)
```